docs/guide/ure.md

```diff
@@ -0,0 +1,70 @@
+
+### UR3e, UR5e and UR10e PROTO
+
+Derived from [Robot](../reference/robot.md).
+
+```
+UR5e/UR5e/UR10e {
+  SFVec3f    translation     0 0 0
+  SFRotation rotation        1 0 0 4.712388966
+  SFString   name            "UR5e"
+  SFString   controller      "void"
+  SFString   controllerArgs  ""
+  SFBool     synchronization TRUE
+  SFBool     selfCollision   TRUE
+  MFNode     toolSlot        []
+  SFBool     staticBase      TRUE
+}
+```
+
+> **File location**: "[WEBOTS\_HOME/projects/robots/universal\_robots/protos/UR3e.proto](https://github.com/omichel/webots/tree/master/projects/robots/universal_robots/protos/UR3e.proto)", "[WEBOTS\_HOME/projects/robots/universal\_robots/protos/UR5e.proto](https://github.com/omichel/webots/tree/master/projects/robots/universal_robots/protos/UR5e.proto)" and "[WEBOTS\_HOME/projects/robots/universal\_robots/protos/UR10e.proto](https://github.com/omichel/webots/tree/master/projects/robots/universal_robots/protos/UR10e.proto)"
+
+#### Field Summary
+
+- `toolSlot`: Extend the robot with new nodes at the end of the arm.
+
+- `staticBase`: Defines if the robot base should be pinned to the static environment.
+
+### Samples
+
+You will find the following sample in this folder: "[WEBOTS\_HOME/projects/robots/universal\_robots/worlds](https://github.com/omichel/webots/tree/master/projects/robots/universal_robots/worlds)".
+
+#### [ure.wbt](https://github.com/omichel/webots/tree/master/projects/robots/universal_robots/worlds/ure.wbt)
+
+![ure.wbt.png](images/robots/ure/ure.wbt.png) This simulation shows an UR3e, an UR5e and an UR10e robots equipped with a [ROBOTIQ 3F Gripper](gripper-actuators.md#robotiq-3f-gripper) grabbing cans on conveyor belts and putting them in crates.
+
+### ROS
+
+To use ROS with the simulated UR5e/UR10e robot in Webots, the `universal_robot_ros` controller should be assigned to the robot.
+Then the `ur_e_webots` ROS package located in "[WEBOTS\_HOME/projects/robots/universal\_robots/ressources/ros\_package/ur\_e\_webots](https://github.com/omichel/webots/tree/master/projects/robots/universal_robots/ressources/ros_package/ur_e_webots)" should be copied into your catkin workspace.
+
+Once `roscore` is started the `ur5e` (replace by `5` by `3` or `10` for the UR3e or UR10e) node of the `ur_e_webots` package can be launched:
+```
+roslaunch ur_e_webots ur5e.launch
+```
+
+The simulation can now be started.
+
+> **Note**: As MoveIt! seems to have difficulties with finding plans for the robot with full joint limits [-2pi, 2pi], there is a joint_limited version using joint limits restricted to [-pi,pi]. In order to use this joint limited version, simply use instead the `ur5e_joint_limited` launch file, i.e.: `roslaunch ur_e_webots ur5e_joint_limited.launch`
+
+#### MoveIt!
+
+Thanks to the [universal\_robot](http://wiki.ros.org/universal_robot) ros package, MoveIt! can be used for motion planning. It can be started with the following launch file:
```

Comment:
```suggestion
Thanks to the [universal\_robot](http://wiki.ros.org/universal_robot) ROS package, MoveIt! can be used for motion planning. It can be started with the following launch file:
```